### DISCUSSION TITLE
Response not received in jmeter.

### DISCUSSION THREAD

**Discussion by Vishwa119** [tilln/jmeter-iso8583#104] [General]
I am sending a request to a switch with header and no trailer. I am using log channel because i dont need any length bit at start, no trailer or padding at end. I can pack and send req to switch and able to receive the succcessful response from switch , which i can see in wireshark. but Jmeter iso8583 plugin couldnt receive it , hence its getting timeout. Also no issue beinglogged like issue in unpacking any fields. Seems all that are okay. I tried using differernt  channels and same is the case. Even tried adding MTI mapping, mux configuration. No Success. What could be the issue.  
![image](https://github.com/user-attachments/assets/f2bf1aa1-adcb-4886-9446-3c70dd780e9a)
![image](https://github.com/user-attachments/assets/d2808182-9a66-40b8-899a-073ebd940aad)
![image](https://github.com/user-attachments/assets/8ed952fa-ed05-4608-9659-b18ce530b09d)
i have put a mock server to test 
![image](https://github.com/user-attachments/assets/5d35a76f-42bc-4dfa-9905-55f917335ada)
![image](https://github.com/user-attachments/assets/a687a722-d95d-48e6-9066-01e6e9ef938f)

Wireshark dump:
port 12015 is the mock server
![image](https://github.com/user-attachments/assets/a9f6f51e-108f-4eb3-93b3-b1a55a6a49bb)

My Observation:
**when Req having field 41 and if response doesnt have field 41 , then response is not received ( Concern !!)**
Additionally , When log channel is used , then response is not received

Appreciate if any input on this..  can DE 41 be bypassed if its an expected behaviour or is it a bug.. please assist in overcoming this..Thanks in Advance!

**Comment by tilln** [comment]
@Vishwa119 
1. Don't use LogChannel.
2. The MTI Mapping is wrong. Remove.
3. If DE41 is not present, the Mux Key Configuration needs to be defined as above with e.g. `11 37`.
4. Try again.

**Reply by Vishwa119** [reply]
Yeah its working Now..Thank you .. But only log channel mimics the message i am having.. without any length bit, trailer , header in ascii... Any way to add custom channel ..

**Reply by tilln** [reply]
If you look at what [LogChannel](https://jpos.org/doc/javadoc/org/jpos/iso/channel/LogChannel.html) does ("Extracts &lt;isomsg&gt; blocks from standard jPOS log") you'll find it is almost certainly not what you are after.

You can of course implement your own custom channel class, by extending [BaseChannel](https://jpos.org/doc/javadoc/org/jpos/iso/BaseChannel.html), though I would recommend you double check that none of the standard channels work.

Are you sure there is no length prefix? Have you confirmed that via a Wireshark trace from your actual switch and/or your messaging specification?
